contracts/plugins/assets/curve/cvx/vendor/ConvexStakingWrapper.sol

```diff
@@ -337,81 +437,55 @@ contract ConvexStakingWrapper is ERC20, ReentrancyGuard {
     }
 
     function _earned(address _account) internal view returns (EarnedData[] memory claimable) {
-        uint256 supply = _getTotalSupply();
-        // uint256 depositedBalance = _getDepositedBalance(_account);
         uint256 rewardCount = rewards.length;
         claimable = new EarnedData[](rewardCount);
 
         for (uint256 i = 0; i < rewardCount; i++) {
             RewardType storage reward = rewards[i];
-
-            //change in reward is current balance - remaining reward + earned
-            uint256 bal = IERC20(reward.reward_token).balanceOf(address(this));
-            uint256 d_reward = bal.sub(reward.reward_remaining);
-
-            //some rewards (like minted cvx) may not have a reward pool directly on the convex pool so check if it exists
-            if (reward.reward_pool != address(0)) {
-                //add earned from the convex reward pool for the given token
-                d_reward = d_reward.add(IRewardStaking(reward.reward_pool).earned(address(this)));
-            }
-
-            uint256 I = reward.reward_integral;
-            if (supply > 0) {
-                I = I + d_reward.mul(1e20).div(supply);
+            if (reward.reward_token == address(0)) {
+                continue;
             }
 
-            uint256 newlyClaimable = _getDepositedBalance(_account)
-            .mul(I.sub(reward.reward_integral_for[_account]))
-            .div(1e20);
-            claimable[i].amount = claimable[i].amount.add(
-                reward.claimable_reward[_account].add(newlyClaimable)
-            );
+            claimable[i].amount = reward.claimable_reward[_account];
             claimable[i].token = reward.reward_token;
-
-            //calc cvx minted from crv and add to cvx claimables
-            //note: crv is always index 0 so will always run before cvx
-            if (i == CRV_INDEX) {
-                //because someone can call claim for the pool outside of checkpoints, need to recalculate crv without the local balance
-                I = reward.reward_integral;
-                if (supply > 0) {
-                    I =
-                        I +
-                        IRewardStaking(reward.reward_pool).earned(address(this)).mul(1e20).div(
-                            supply
-                        );
-                }
-                newlyClaimable = _getDepositedBalance(_account)
-                .mul(I.sub(reward.reward_integral_for[_account]))
-                .div(1e20);
-                claimable[CVX_INDEX].amount = CvxMining.ConvertCrvToCvx(newlyClaimable);
-                claimable[CVX_INDEX].token = cvx;
-            }
         }
         return claimable;
     }
 
     function claimRewards() external {
         uint256 cvxOldBal = IERC20(cvx).balanceOf(msg.sender);
         uint256 crvOldBal = IERC20(crv).balanceOf(msg.sender);
-        _checkpointAndClaim([address(msg.sender), address(msg.sender)]);
+        address _account = address(msg.sender);
```

Comment:
If we're implementing a reward redirect, the CVX and CRV calculations must happen against the reward target as well.

Reply:
Thats what the code is doing. Whats your suggested change?

Reply:
Look at the `_checkpointAndClaim` call within `claimRewards()

Reply:
The CRV and CVX calculations for the emitted events, I just pushed a change to this branch, let me know if that makes sense or we can change it further.

Reply:
Ok gotcha.. yeah.. looks good